testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/authentication/actions/TermsAndConditionsTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2013 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.testsuite.console.authentication.actions;
+
+import java.util.concurrent.TimeUnit;
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.testsuite.auth.page.login.TermsAndConditions;
+import org.keycloak.testsuite.console.AbstractConsoleTest;
+import org.keycloak.testsuite.console.page.authentication.RequiredActions;
+import org.keycloak.testsuite.console.page.users.User;
+import org.keycloak.testsuite.console.page.users.CreateUser;
+
+import org.keycloak.testsuite.console.page.users.UserAttributesForm;
+import org.keycloak.testsuite.util.WaitUtils;
+
+/**
+ *
+ */
+public class TermsAndConditionsTest extends AbstractConsoleTest {
+
+    private static final String USERNAME = "user1";
+    
+    private static final String PASSWORD = "user1";
+    
+    private static final String TERMS_TEXT = "Terms and conditions to be defined";
+    
+    @Page
+    private RequiredActions requiredActionsPage;
+
+    @Page
+    private TermsAndConditions termsAndConditionsPage;
+
+    @Page
+    CreateUser createUser;
+    
+    @Page
+    User user;
+
+    /**
+     * Increase 'implicitWait'. If smaller then logout from Test Realm Admin Console
+     * may not succeed.
+     */
+    @Override
+    protected void driverSettings() {
+        driver.manage().timeouts().pageLoadTimeout(WaitUtils.PAGELOAD_TIMEOUT, TimeUnit.MILLISECONDS);
+        driver.manage().timeouts().implicitlyWait(10000, TimeUnit.MILLISECONDS);
+        driver.manage().window().maximize();
+    }
+    
+    @Before
+    public void createUserAccount() throws Exception {
```

Comment:
You need to test 3 different scenarios for terms and conditions (T&C), here you are only testing one. The scenarios to test are:

1. User exists prior to enabling terms and conditions. First login to check that the is not asked to confirm T&C, then add the required action through admin console and check the user is required
2. User self-registers after T&C is enabled. Check that user is required to confirm T&C during registration.
3. User is registered by admin. Check that user is required to confirm T&C on first login (this is what you are testing now).

Finally you should not use the admin console (RequiredActions, CreateUser) to configure tests. Use the admin endpoints. Admin console approach is slow and unstable, which is why we've separated the admin console tests out.